client/state/posts/actions.js

```diff
@@ -296,3 +298,18 @@ export function restorePost( siteId, postId ) {
 		} );
 	};
 }
+
+/**
+ * Returns an action thunk which, when dispatched, triggers a network request
+ * to create a new term. All actions dispatched by the thunk will include meta
+ * to associate it with the specified post ID.
+ *
+ * @param  {Number}   siteId   Site ID
+ * @param  {String}   taxonomy Taxonomy Slug
+ * @param  {Object}   term     Object of new term attributes
+ * @param  {Number}   postId   ID of post to which term is associated
+ * @return {Function}          Action thunk
+ */
+export function addTermForPost( siteId, taxonomy, term, postId ) {
+	return dispatchWithMeta( addTerm( siteId, taxonomy, term ), { postId } );
```

Comment:
Is the main purpose here to compose the existing action with new data so as not to repeat code? Given an existing pure action it's clear how we could skip all this with `Object.assign()`, but I see how the thunk mucks it all up.

Otherwise, we're looking at writing a _second_ `addTerm()` except that it includes the `postId`, right?